### DISCUSSION TITLE
设置指定边界的初始深度

### DISCUSSION THREAD

**Discussion by Kong-JiaTao** [SynxFlow/SynxFlow#10] [Q&A] [answered]
![Snipaste_2024-07-16_11-01-36](https://github.com/user-attachments/assets/323401e5-cfa4-42a9-bc6b-a593d4bbbf13)
我只想要设置河道的水位,不想设置整个DEM的水深

**Comment by Kong-JiaTao** [comment]
case_input.set_initial_condition('h0', 1.0) 当前设置1的水深,设置后的图
![h_0](https://github.com/user-attachments/assets/40efb31b-306c-44a7-8bf4-4ee348cba65c)

**Comment by Kong-JiaTao** [comment]
case_input.set_initial_condition('h0', 1.0) 第二个参数用numpy

**Comment by mingxiaodong** [comment, marked as answer]
set_initial_condition 是用来设置初始条件的，设置边界条件用的是set_boundary_condition
可以参考我们的[洪水样例](https://synxflow.readthedocs.io/en/latest/Tutorials/flood.html#Generating-the-inputs-and-runing-the-simulation)代码8-10段

**Comment by xiaxilin** [comment]
To set up non-uniform initial conditions, you can use a raster. For example

```Python
h = IO.Raster('path/to/raster')
case_input.set_initial_condition('h0', h.array)
```
